postreise/analyze/generation/summarize.py

```diff
@@ -129,3 +133,42 @@ def summarize_hist_gen(hist_gen_raw: pd.DataFrame, all_resources: list) -> pd.Da
     result.rename(columns=type2label, inplace=True)
 
     return result
+
+
+def get_generation_time_series_by_resources(scenario, area, resources, area_type=None):
+    """Get time series generation for each resource in certain area of a scenario
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources
+    :param str area_type: one of: *'loadzone'*, *'state'*,
+        *'state_abbr'*, *'interconnect'*
+    :return: (*pandas.DataFrame*) -- data frame of time series generation for each
+        resource, index: time stamps, columns: resources
+    """
+    _check_scenario_is_in_analyze_state(scenario)
+    plant_id = get_plant_id_for_resources_in_area(
+        scenario, area, resources, area_type=area_type
+    )
+    pg = scenario.state.get_pg()[plant_id]
+    grid = scenario.state.get_grid()
+
+    return pg.T.groupby(grid.plant["type"]).agg(sum).T
```

Comment:
This seems to be equivalent to `pg.groupby(grid.plant.type, axis=1).sum()`

Reply:
Yeah, good catch. I simply copied this line from `analyze_pg`. Will simplify it as you suggested.

Reply:
It takes me some time to figure it out. In this case, current line will work: `pg.groupby(grid.plant.loc[plant_id, "type"].values, axis=1).sum()`.

Reply:
Ah, we need the list we pass to the groupby to have the same length as the filtered pg?

Reply:
Exactly, unless we pass function/mapping rule otherwise it won't auto interpreted in the way when `groupby` with default axis.

Reply:
Done.